### ISSUE TITLE
Compile error when IEnumerable<T> properties where T is a complex reference type

### ISSUE THREAD

**Issue by colgreen** [ladeak/JsonMergePatch#68]
Let's say we have a DTO set-up like so:

```csharp
[Patchable]
public class Client
{
    public string? Forename { get; set; }
    public IEnumerable<ClientAddress>? Addresses { get; set; }
}

[Patchable]
public class ClientAddress
{
   public string? Line1 { get; set; }
   public string? City { get; set; }
}
```

The generated wrapper class code contains this line:

ApplyChanges()
```csharp
            if (Properties[1])
                input.Addresses = Addresses;
```

Which results in the following compiler error:
```
CS0266	Cannot implicitly convert type 'System.Collections.Generic.IEnumerable<LaDeak.JsonMergePatch.Generated.SafeFoo.ClientAddressWrapped>' to 'System.Collections.Generic.IEnumerable<Foo.ClientAddress>'. An explicit conversion exists (are you missing a cast?)
```

I.e., ClientAddressWrapped is a sub-type of ClientAddress, and therefore a list or IEnumerable of ClientAddressWrapped  can be cast to ClientAddress (as the lists contain references/pointers to the reference types on the heap), but only as an explicit cast.

**Comment by ladeak** [comment]
https://github.com/ladeak/JsonMergePatch#use-maps

Generally arrays, enumerables were supported as above.

Let me think about this use case though.

**Comment by ladeak** [comment]
We can think about this use case though.

**Comment by ladeak** [comment]
What would be your expected behavior for patching an `IEnumerable<T>` ?

**Comment by colgreen** [comment]
I think it's generally understood that the entire JSON array is replaced - i.e., it's all or nothing. It's called out in [RFC 7396](https://datatracker.ietf.org/doc/html/rfc7396)...

> There are a few things to note about the function.  If the patch is
>    anything other than an object, the result will always be to replace
>    the entire target with the entire patch.  Also, it is not possible to
>    patch part of a target that is not an object, such as to replace just
>    some of the values in an array.

**Comment by ladeak** [comment]
I am looking into this.

**Comment by colgreen** [comment]
I think the type used for 'array' properties on the wrapper classes can be List<T>, if that's what you're asking.  I think that would be fine. I don't foresee any problems with that approach.

**Comment by ladeak** [comment]
Supporting `Lists<T>` - Released with v6.1.0
